src/modules/Account/Devices.ts

```diff
@@ -222,6 +230,47 @@ class Devices extends TagoIOModule<GenericModuleParams> {
 
     return result;
   }
+
+  /**
+   * Get data from all variables in the device.
+   *
+   * @param deviceId Device ID.
+   * @param queryParams Query parameters to filter the results.
+   *
+   * @returns Array with the data values stored in the device.
+   *
+   * @example
+   * ```ts
+   * const myDevice = new Device({ token: "my_device_token" });
+   *
+   * const lastTenValues = await myDevice.getVariablesData("myDeviceId", { qty: 10 });
+   * ```
+   */
+  public async getDeviceData(deviceId: GenericID, queryParams?: DataQuery): Promise<Data[]> {
```

Comment:
New route to get device data with profile token for the `Variables` tab in next generation Device types.